functionary/builder/models/build.py

```diff
@@ -56,6 +56,26 @@ class Meta:
     def __str__(self):
         return f"{self.id}"
 
+    def error(self):
+        """Saves status as `ERROR`"""
+        self._update_status(self.ERROR)
+
+    def pending(self):
+        """Saves status as `PENDING`"""
+        self._update_status(self.PENDING)
+
+    def complete(self):
+        """Saves status as `COMPLETE`"""
+        self._update_status(self.COMPLETE)
+
+    def in_progress(self):
+        """Saves status as `IN_PROGRESS`"""
+        self._update_status(self.IN_PROGRESS)
+
+    def _update_status(self, status: str) -> None:
+        self.status = status
+        self.save()
```

Comment:
Isn't this going to also save any other unsaved modifications to the build object?

Reply:
Yes. I can modify it to only save the status field. The other update status methods are just calling `self.save()`, so maybe I'll go back and update them to just save the status field in another PR

Reply:
Oh no, definitely don't do that.  The confusion that will be caused by having an object that is only partially consistent with the state of the database seems way more likely to be harmful than helpful.

Reply:
Updated to just perform the save on the `status` field